packages/Datadog.Unity/Tests/Rum/DatadogTrackedWebRequestTest.cs

```diff
@@ -122,6 +123,91 @@ public void GeneratesCorrectDatadogAttributesWhenSampled()
             Assert.AreEqual(_sampled ? 1.0f : 0.0f, attributes["_dd.rule_psr"]);
         }
 
+        [TestCase(TracingHeaderType.Datadog)]
+        [TestCase(TracingHeaderType.B3)]
+        [TestCase(TracingHeaderType.B3Multi)]
+        [TestCase(TracingHeaderType.TraceContext)]
+        public void DatadogAttributesAndTracingHeadersHaveSameValue(TracingHeaderType headerType)
+        {
+            // Given
+            var headers = new Dictionary<string, string>();
+            var attributes = new Dictionary<string, object>();
+            var context = _trackingHelper.GenerateTraceContext();
+
+            // When
+            _trackingHelper.GenerateDatadogAttributes(context, attributes);
+            _trackingHelper.GenerateTracingHeaders(context, headerType, headers);
+
+            // Then
+            var traceString = attributes.GetValueOrDefault("_dd.trace_id")?.ToString();
+            var spanString = attributes.GetValueOrDefault("_dd.span_id")?.ToString();
+            if (_sampled)
+            {
+                Assert.IsNotNull(traceString);
+                BigInteger.TryParse(traceString, NumberStyles.HexNumber, null, out var attributeTraceId);
+
+                Assert.IsNotNull(spanString);
+                BigInteger.TryParse(spanString, out var attributeSpanId);
+
+                GetIdsFromHeaders(headers, headerType, out var headerTraceId, out var headerSpanId);
+                Assert.AreEqual(headerTraceId, attributeTraceId);
+                Assert.AreEqual(headerSpanId, attributeSpanId);
+            }
+            else
+            {
+                Assert.IsNull(traceString);
+                Assert.IsNull(spanString);
+            }
+
+            Assert.AreEqual(_sampled ? 1.0f : 0.0f, attributes["_dd.rule_psr"]);
+        }
+
+        private void GetIdsFromHeaders(Dictionary<string, string> headers, TracingHeaderType headerType, out BigInteger traceId, out BigInteger spanId)
+        {
+            switch (headerType)
+            {
+                case TracingHeaderType.Datadog:
+                {
+                    BigInteger.TryParse(headers["x-datadog-trace-id"], out traceId);
+                    BigInteger.TryParse(headers["x-datadog-parent-id"], out spanId);
```

Comment:
Actually, since `GetIdsFromHeaders` is using `out BigInteger` it's easier to parse them here as BigInteger than to `ulong`. The performance is probably slightly worse but I'm not really concerned about it for the test.